### PR TITLE
Support patches located in the RootPackage

### PR DESCRIPTION
**Pull request by KorvinSzanto**
This allows us to do the following:

1. Declare patches on the root package:
```
"extra": {
    "patches": {
        "thingto/patch": {
            "Patch Description": "relative/path/to/patch.patch"
         }
    }
}
```
2. Create a patch file at `./relative/path/to/patch.patch`
3. Run composer install

The reason someone might want to do this is it eliminates the need to have patches located in a separate composer repository. So if you need to make a patch to a single repository you avoid a few steps.

### CODE REVIEW

File: src/PatchCollector.php [hunk 1]
```diff
@@ -143,7 +143,12 @@ protected function resolveExplicitList(PackageInterface $package, $patches)
         if (!is_array($patches)) {
             $this->handleException(Exception\InvalidPackageConfigurationValue($package, 'extra.patches', $patches, 'The extra.patches configuration must be an array.'));
         } else {
-            $packageDirectory = $this->installationManager->getInstallPath($package);
+            // If the package is the RootPackage, we won't be able to get its install path from the installationmanager
+            if ($package instanceof RootPackageInterface) {
+                $packageDirectory = getcwd();
```
Comment (KorvinSzanto) [inline src/PatchCollector.php:148]: I wasn't able to find a better way to resolve the path of the RootPackage instance.

**Comment by mlocati** [comment]
Does this work also when `composer` is executed using the `--working-dir` option?

**Comment by KorvinSzanto** [comment]
It has with my testing, here's the dump of `getcwd()` when the `--working-dir` option is being used:
```
$ ~/projects/concrete5 ❯ composer install --working-dir=composer                                                                                                                                                                                                                                                                                          130  
Gathering patches from concrete5/composer (extra.patches).
string(41) "/Users/korvin/projects/concrete5/composer"
```

**Comment by mlocati** [comment]
Included in new release: https://github.com/mlocati/composer-patcher/releases/tag/1.1.0

**Pull request merged**